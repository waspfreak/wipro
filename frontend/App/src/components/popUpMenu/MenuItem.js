import React from 'react';
import { Text, View, TouchableWithoutFeedback } from 'react-native';
import styles from './styles';
import PropTypes from 'prop-types';
import Icon from '../icon/Icon';
import colors from '../../styles/colors';

export const MenuItem = ({ title, onPress, iconName, iconColor, last, testIdentifier }) => {
  return (
    <TouchableWithoutFeedback style={styles.menuItem} onPress={onPress} testID={testIdentifier}>
      <View style={styles.itemContainer}>
        <View style={styles.iconContainer}>
          <Icon name={iconName} fill={iconColor} height="25" width="25" />
        </View>
        <View style={styles.textMenuItem}>
          <Text>{title}</Text>
          <View
            style={last ? styles.dividerMenuItemLast : styles.dividerMenuItem}
          />
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

MenuItem.propTypes = {
  title: PropTypes.string,
  onPress: PropTypes.func,
  iconName: PropTypes.string,
  iconColor: PropTypes.string,
  last: PropTypes.bool
};

MenuItem.defaultProps = {
  iconColor: colors.blue
};

export default MenuItem;
