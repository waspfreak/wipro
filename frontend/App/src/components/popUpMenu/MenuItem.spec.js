import React from 'react';
import { render } from '@testing-library/react-native';
import MenuItem from './MenuItem';

describe('MenuItem tests', () => {
  it('should match MenuItem snapshot', () => {
    const { baseElement } = render(<MenuItem />);
    expect(baseElement).toMatchSnapshot();
  });
});
