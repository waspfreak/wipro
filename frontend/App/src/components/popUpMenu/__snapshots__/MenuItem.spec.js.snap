// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem tests should match MenuItem snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableWithoutFeedback
      style={
        Object {
          "width": 675,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingLeft": 10,
            "paddingRight": 0,
          }
        }
      >
        <View
          style={
            Object {
              "paddingRight": 10,
              "paddingTop": 4,
            }
          }
        />
        <View
          style={
            Object {
              "color": "#000000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 20,
              "paddingTop": 8,
              "width": "85%",
            }
          }
        >
          <Text />
          <View
            style={
              Object {
                "borderBottomColor": "#aba7a0",
                "borderBottomWidth": 1,
                "flex": 1,
                "marginTop": 9,
              }
            }
          />
        </View>
      </View>
    </TouchableWithoutFeedback>
  </View>
</View>
`;
